src/youtube/js/youtube.js

```diff
@@ -0,0 +1,374 @@
+          element.raiseBitratesAvailable();
+          bitrateFlag = false;
+        }      
+        break;
+      case 2:
+        // paused 
+        element.controller.notify( element.controller.EVENTS.PAUSED);
+        break;
+      case 3:
+        // buffering
+        OO.log("Youtube: Buffering event received"); 
+        break;
+      case 5:
+        // video cued 
+        break;
+    }
+  };
+
+  /*
+   *The Youtube iframe API calls this function when the player's throws an error.
+   */
+  function onPlayerError(event) {
+    var code = -1;
+    switch(event.data) {
+      case 2:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // invalid video id
+        break;
+      case 5:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // The requested content cannot be played in an HTML5 player or another error related to the HTML5 player has occurred
+        break;
+      case 100:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // The video requested was not found. This error occurs when a video has been removed (for any reason) or has been marked as private 
+        break;
+      case 101:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // The owner of the requested video does not allow it to be played in embedded players. 
+        break;
+      case 150:
+        element.controller.notify(element.controller.EVENTS.ERROR, { "errorcode" : code });
+        // This error is the same as 101. It's just a 101 error in disguise! 
+        break;
+      }
+  };
+
+  /**
+   * @class OoyalaYoutubeVideoWrapper
+   * @classdesc Player object that wraps the video element.
+   * @property {object} controller A reference to the Ooyala Video Tech Controller
+   * @property {boolean} disableNativeSeek When true, the plugin should supress or undo seeks that come from
+   *                                       native video controls
+   */  
+  var OoyalaYoutubeVideoWrapper = function() {
+    this.controller = {};
+    this.disableNativeSeek = false;
+    var timeUpdateInterval = null;
+    var hasPlayed = false;
+    var videoEnded = false;
+
+    /**
+     * Triggers playback on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#play
+     */
+    this.play = function() {
+      if(playerReady)
+      {
+        youtubePlayer.playVideo();
+        this.controller.notify(this.controller.EVENTS.PLAY, { url: youtubeID });
+        updateTimerDisplay();
+        hasPlayed = true;
+      }
+      else {
+        javascriptCommandQueue.push(["play", null]);
+        playerNotready = false;
+      }
+    };
+
+    /**
+     * Triggers a pause on the video element.
+     * @public
+     * @method OoyalaYoutubeVideoWrapper#pause
+     */
+    this.pause = function(){
+      youtubePlayer.pauseVideo();
```

Comment:
youtubePlayer null check. If it doesn't exist, then exit early and give a debug warning